libs/packages/layouts/src/lib/feature/download-modal/download-modal.component.ts

```diff
@@ -0,0 +1,172 @@
+      summaryInfo: new FormControl(false, Validators.required),
+      caseRecords: new FormControl(false, Validators.required),
+      currentRecord: new FormControl(false, Validators.required),
+      savedSearch: new FormControl(false, Validators.required),
+      fileType: new FormControl('', Validators.required)
+    });
+
+    this.model = {
+      name: '',
+      fullRecord: false,
+      summaryInfo: false,
+      caseRecords: false,
+      currentRecord: false,
+      savedSearch: true,
+      fileType: ''
+
+    };
+
+    this.fields = [
+      {
+        key: 'fullRecord',
+        type: 'checkbox',
+        templateOptions: {
+          label: 'Download full record.',
+          required: true
+        }
+      },
+      {
+        key: 'summaryInfo',
+        type: 'checkbox',
+        templateOptions: {
+          label: 'Download summary information only.',
+          required: true
+        }
+      },
+      {
+        key: 'caseRecords',
+        type: 'checkbox',
+        templateOptions: {
+          label: 'Download FDPS case records.',
+          required: true
+        }
+      },{
+        key: 'currentRecord',
+        type: 'checkbox',
+        templateOptions: {
+          label: 'Download current record and all of its modification.',
+          required: true
+        }
+      },
+      {
+        key: 'fileType',
+        type: 'radio',
+        templateOptions: {
+          label: 'Select File Type',
+          required: true,
+          options: [
+            {
+              key: 'csv',
+              value: 'CSV File'
+            },
+            {
+              key: 'pdf',
+              value: 'PDF File'
+            },
+            {
+              key: 'zip',
+              value: 'Zip File'
+            }
+          ]
+
+        }
+
+      },
+      {
+        key: 'name',
+        type: 'input',
+        templateOptions: {
+          required: true,
+          label: 'Name'
+        }
+      },
+      {
+        key: 'savedSearch',
+        type: 'checkbox',
+        templateOptions: {
+          label: 'Add to my Saved Searches.',
+          required: false
+        }
+      }
+    ];
+
+    this.options = {};
+  }
+  onSubmit(){
+    //console.log(this.model);
+    //this.onFormGroupChange.emit(this.model);
+
+    // this.httpclient.get(url of the actual file)
+    of('filecontent').subscribe(
+      fileContent => {
+        let fileMimeType = 'text/plain';
+
+        switch(this.model.fileType) {
+          case 'pdf':
+            fileMimeType = '/downloads/archive.pdf';
+            break;
+          case 'csv':
+            fileMimeType = '/downloads/archive.csv';
+          case 'zip':
+            fileMimeType = '/downloads/archive.zip';
+
+        }
+
+
+        const fileBlob = new Blob([fileContent], {type: fileMimeType});
+        const a = document.createElement('a');
+        a.download = name;
+        a.href = window.URL.createObjectURL(fileBlob);
+        a.click();
+      }
+    );
```

Comment:
Being a common component we should leave all of the logic of the download up to the implementing developer. The onSubmit/outputs should just make the formly model available to them.